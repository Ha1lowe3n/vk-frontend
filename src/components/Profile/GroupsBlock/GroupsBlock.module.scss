.groupsBlock {
    .groups {
        padding: 12px 15px 15px;
    }
}
