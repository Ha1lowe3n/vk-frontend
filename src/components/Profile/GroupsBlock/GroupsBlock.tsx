import React from "react";
import { WhiteBlockInfo } from "../../WhiteBlock/WhiteBlockInfo/WhiteBlockInfo";
import { Group } from "./Group/Group";

import styles from "./GroupsBlock.module.scss";

const dataGroups = [
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        title: "Vladilen",
        description: "hello",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        title: "Vladilen",
        description: "hello",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        title: "Vladilen",
        description: "hello",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        title: "Vladilen",
        description: "hello",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        title: "Vladilen",
        description: "hello",
    },
];

export const GroupsBlock: React.FC = () => {
    return (
        <div className={styles.groupsBlock}>
            <WhiteBlockInfo title={"Following"} count={60} />
            <div className={styles.groups}>
                {dataGroups.map((group, i) => (
                    <Group
                        key={i}
                        style={i > 0 ? { marginTop: "10px" } : undefined}
                        title={group.title}
                        description={group.description}
                        groupAvatar={group.src}
                    />
                ))}
            </div>
        </div>
    );
};
