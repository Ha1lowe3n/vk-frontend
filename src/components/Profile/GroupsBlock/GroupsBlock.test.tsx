import { render, screen } from "@testing-library/react";
import { GroupsBlock } from "./GroupsBlock";

it("render GiftsBlock component", () => {
    render(<GroupsBlock />);
    expect(screen.getByText("Following")).toBeInTheDocument();
    expect(screen.getByText("60")).toBeInTheDocument();
    expect(screen.getAllByAltText("group avatar")[0]).toBeInTheDocument();
});

it("GiftsBlock snapshot", () => {
    const view = render(<GroupsBlock />);
    expect(view).toMatchSnapshot();
});
