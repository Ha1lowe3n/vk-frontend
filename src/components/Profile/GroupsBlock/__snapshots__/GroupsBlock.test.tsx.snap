// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftsBlock snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="groupsBlock"
      >
        <div
          class="whiteBlockInfo"
        >
          <span
            class="title"
          >
            Following
          </span>
          <span
            class="count"
          >
            60
          </span>
        </div>
        <div
          class="groups"
        >
          <div
            class="group"
          >
            <a
              href="#"
            >
              <img
                alt="group avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <div
              class="text"
            >
              <span
                class="title"
              >
                <a
                  class="link_text"
                  href="#"
                >
                  Vladilen
                </a>
              </span>
              <span
                class="description"
              >
                hello
              </span>
            </div>
          </div>
          <div
            class="group"
            style="margin-top: 10px;"
          >
            <a
              href="#"
            >
              <img
                alt="group avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <div
              class="text"
            >
              <span
                class="title"
              >
                <a
                  class="link_text"
                  href="#"
                >
                  Vladilen
                </a>
              </span>
              <span
                class="description"
              >
                hello
              </span>
            </div>
          </div>
          <div
            class="group"
            style="margin-top: 10px;"
          >
            <a
              href="#"
            >
              <img
                alt="group avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <div
              class="text"
            >
              <span
                class="title"
              >
                <a
                  class="link_text"
                  href="#"
                >
                  Vladilen
                </a>
              </span>
              <span
                class="description"
              >
                hello
              </span>
            </div>
          </div>
          <div
            class="group"
            style="margin-top: 10px;"
          >
            <a
              href="#"
            >
              <img
                alt="group avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <div
              class="text"
            >
              <span
                class="title"
              >
                <a
                  class="link_text"
                  href="#"
                >
                  Vladilen
                </a>
              </span>
              <span
                class="description"
              >
                hello
              </span>
            </div>
          </div>
          <div
            class="group"
            style="margin-top: 10px;"
          >
            <a
              href="#"
            >
              <img
                alt="group avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <div
              class="text"
            >
              <span
                class="title"
              >
                <a
                  class="link_text"
                  href="#"
                >
                  Vladilen
                </a>
              </span>
              <span
                class="description"
              >
                hello
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="groupsBlock"
    >
      <div
        class="whiteBlockInfo"
      >
        <span
          class="title"
        >
          Following
        </span>
        <span
          class="count"
        >
          60
        </span>
      </div>
      <div
        class="groups"
      >
        <div
          class="group"
        >
          <a
            href="#"
          >
            <img
              alt="group avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <div
            class="text"
          >
            <span
              class="title"
            >
              <a
                class="link_text"
                href="#"
              >
                Vladilen
              </a>
            </span>
            <span
              class="description"
            >
              hello
            </span>
          </div>
        </div>
        <div
          class="group"
          style="margin-top: 10px;"
        >
          <a
            href="#"
          >
            <img
              alt="group avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <div
            class="text"
          >
            <span
              class="title"
            >
              <a
                class="link_text"
                href="#"
              >
                Vladilen
              </a>
            </span>
            <span
              class="description"
            >
              hello
            </span>
          </div>
        </div>
        <div
          class="group"
          style="margin-top: 10px;"
        >
          <a
            href="#"
          >
            <img
              alt="group avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <div
            class="text"
          >
            <span
              class="title"
            >
              <a
                class="link_text"
                href="#"
              >
                Vladilen
              </a>
            </span>
            <span
              class="description"
            >
              hello
            </span>
          </div>
        </div>
        <div
          class="group"
          style="margin-top: 10px;"
        >
          <a
            href="#"
          >
            <img
              alt="group avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <div
            class="text"
          >
            <span
              class="title"
            >
              <a
                class="link_text"
                href="#"
              >
                Vladilen
              </a>
            </span>
            <span
              class="description"
            >
              hello
            </span>
          </div>
        </div>
        <div
          class="group"
          style="margin-top: 10px;"
        >
          <a
            href="#"
          >
            <img
              alt="group avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <div
            class="text"
          >
            <span
              class="title"
            >
              <a
                class="link_text"
                href="#"
              >
                Vladilen
              </a>
            </span>
            <span
              class="description"
            >
              hello
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
